server/src/test/java/org/elasticsearch/search/SearchHitsTests.java

```diff
@@ -143,4 +159,23 @@ public void testToXContent() throws IOException {
             "\"hits\":[{\"_type\":\"type\",\"_id\":\"id1\",\"_score\":null},"+
             "{\"_type\":\"type\",\"_id\":\"id2\",\"_score\":null}]}}", Strings.toString(builder));
     }
+
+    public void testFromXContentWithShards() throws IOException {
```

Comment:
sorry to be a pain: could we have a test that does not randomize stuff and just exercises the _shard scenario? I think otherwise, this test kind of does what testFromXContent does, but it does not assert on equality rather than testing something specifically that is otherwise not covered.